scripts/noaa/gpcc_spi/gpcc_spi_aggregation.py

```diff
@@ -0,0 +1,130 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pandas as pd
+import json
+from typing import List
+import logging
+from datetime import datetime
+import os
+import csv
+import glob
+from pathlib import Path
+
+from absl import flags
+from absl import app
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string('in_pattern', '/tmp/gpcc_spi/out/*.csv',
+                    'The directory the csvs are.')
+flags.DEFINE_string('place_area_ratio_json_path',
+                    'scripts/noaa/gpcc_spi/geojson_data/place_area_ratio.json',
+                    'Path to place area json cache.')
+flags.DEFINE_string('output_dir', '/tmp/gpcc_spi/agg',
+                    'The directory where the output mcf will be generated in.')
+
+
+def get_place_grid_ratio_dict(path):
+    with open(path, 'r') as f:
+        return json.load(f)
+
+
+# TODO(alexyfchen): Improve with df operations rather than loop.
+def aggregate_with_loop(output_dir, place_grid_ratio_dict, spi_df, period):
+    path = os.path.join(output_dir, f'gpcc_spi_pearson_{period}_agg.csv')
+    if os.path.isfile(path):
+        os.remove(path)
+
+    columns = ['place', 'time', 'variable', 'spi']
+    with open(path, 'w') as f:
+        f.write(','.join(columns) + '\n')
+
+    for place_id, grids in place_grid_ratio_dict.items():
+        # Find all spis relevant to the grids
+        relevant_spis = spi_df.loc[spi_df['grid_key'].isin(
+            [g["grid"] for g in grids])]
+        if len(relevant_spis) == 0:
+            continue
+
+        place_df = pd.DataFrame([[g['grid'], g['ratio']] for g in grids],
+                                columns=['grid_key', 'ratio'])
+        place_df.set_index('grid_key')
+
+        merged = relevant_spis.merge(place_df)
+        merged['spi'] = merged['spi'] * merged['ratio']
+
+        weighted = merged.groupby('time').agg(sum)
+
+        filtered = weighted.loc[weighted['ratio'] > 0.999].reset_index()
+
+        time_weighted_spi = filtered.drop('ratio', axis=1)
+        time_weighted_spi['place'] = place_id
+        time_weighted_spi[
+            'variable'] = f"dcs:standardizedPrecipitationIndex_Atmosphere_{int(period)}MonthPeriod"
+
+        time_weighted_spi.to_csv(path,
+                                 columns=columns,
+                                 header=False,
+                                 mode='a',
+                                 index=False)
+        return path
+
+
+def _read_spi_data_csv(csv_path):
+
+    def grid_to_coord(grid):
+        """grid_1/lat_lng -> lat^lng"""
+        lat_lng = grid.split('/')[1]
+        return "^".join(lat_lng.split('_'))
+
+    df = pd.read_csv(csv_path, usecols=['time', 'place', 'spi'])
```

Comment:
How big are the CSVs?  For large ones I tend to prefer not doing pandas and instead looping over rows.